qualcommax: ipq807x: add support for Netgear WAX630

skramstad opened the pull request:
```
Specifications:
* CPU: Qualcomm IPQ8074A, SoC Version: 2.0, Quad core 1651 MHz
* RAM: 1 GiB of DDR3 466 MHz
* Flash: NAND 512 MiB (Winbond W29N04GZ)
* 6 RGB LEDs: Power, LAN1, LAN2, 2.4GHz, 5GHz H and 5GHz L
* UART: One 4-pin populated header next to the heatsink and a chip.
  GND RXD TXD, beginning from the external antennas. 115200n8.

Lan:
* One 100/1000/2.5GBASE-T Gigabit Ethernet 802.3bt/at
* One 100/1000 Gigabit Ethernet

Wlan:
* 4x4 in 2.4GHz  : 802.11b/g/n/ax
* 4x4 in 5.0GHz L: 802.11a/n/ac/ax
* 4x4 in 5.0GHz H: 802.11a/n/ac/ax
* OFDM and OFDMA
* Bidir and MU-MIMO
* Internal antenna 2.86/4.41/4.98 dBi (2.4GHz/5GHz L/5GHz H)

Power:
* 802.3bt/at 30.1W
* DC 12V/3.5A

Mounting: Wall and ceiling
```

```
1.  Download the OpenWrt initramfs image. Copy the image to a TFTP server
2.  Connect to the console on the AP, and connect the LAN port to your LAN
3.  Stop auto boot to get to U-boot shell, interrupt the autoboot process by pressing '0' when prompted
4.  Set active_fw in env
    # setenv active_fw 1
5.  Transfer the initramfs image with TFTP
    # setenv serverip 192.168.1.10 (IP of TFTP server host)
    # setenv ipaddr 192.168.1.1 (IP used by the router for getting the image, must be in the same subnet as the TFTP host)
    # tftpboot openwrt-qualcommax-ipq807x-netgear_wax630-initramfs-uImage.itb
6.  Reboot and load the image
    # bootm
7.  SCP factory image to the AP
    # scp openwrt-qualcommax-ipq807x-netgear_wax630-squashfs-factory.ubi root@192.168.1.1:/tmp/
8.  Connect to device using SSH (use the LAN port)
9.  Flash squashfs-factory.ubi from within the initramfs instance of OpenWRT
    Before you flash, please check your mtd partitions where mtdX is the right mtd rootfs partition.
    # cat /proc/mtd (To check MTD partitions)
    # ubiformat /dev/mtd18 -y -f /tmp/openwrt-qualcommax-ipq807x-netgear_wax630-squashfs-factory.ubi
10. Set active_fw to 0
    # /usr/sbin/fw_setenv active_fw 0
11. Reboot the AP and your done
    # reboot
```

Signed-off-by: Kristian Skramstad <kristian+github@83.no>

skramstad commented:
Note: I did change the order of the wax-family in the image/ipq807x.mk file. Since Robi last time wanted 218, 620 and 630. Hope it's okey.

castiel652 commented:
> Note: I did change the order of the wax-family in the image/ipq807x.mk file. Since Robi last time wanted 218, 620 and 630. Hope it's okey.

Maybe make that change into a separate commit?

skramstad commented:
> > Note: I did change the order of the wax-family in the image/ipq807x.mk file. Since Robi last time wanted 218, 620 and 630. Hope it's okey.
> 
> Maybe make that change into a separate commit?

Sure thing. But I don't make that call, so I'll leave it to the them. 😃

robimarko commented:
Yeah, please separate the reordering into a separate commit

skramstad commented:
Okey. Reversed the order and added back wax630 below wax620.

castiel652 commented:
You can still put the seperate commit in the same PR.

skramstad commented:
> You can still put the seperate commit in the same PR.

Is it okey now?

robimarko commented:
You want to do the sorting before adding the new device, also prefix the commits with `qualcommax: ipq807x:`

skramstad commented:
Whatever I do to make it right, like commit the sorting before etc just mess up the whole thing.

But as you can see, git is not my area so I don't know how to fix this. I'm sorry!

castiel652 commented:
Take it easy. I know it's complicated. Don't close the PR because of this. :)

Let's fix it steps by steps.
Assuming you are using the CLI.

1. First let's make a backup  branch for this current PR.  `git switch -C backup`.
now `git status` should show that you are on a branch called `backup`.
and `git log` should contain these two commits.
2. `git checkout main` to switch back to the `main` branch
3. `git reset HEAD~2` this should discard the last two commits but still preserve the changes.
4.  sort `ipq807x.mk` and `01_leds` but remove the part with WAX630
5.  stage and commit (don't forget to use `--sign-off`) with the title `qualcommax: ipq807x: sort WAX family in Makefile and LED` (or something else if Robi has a better idea for the title)
6. add WAX630 to the `ipq807x.mk` and `01_leds`
7. stage all your changes for WAX630 and commit.

skramstad commented:
Thanks @castiel652. I followed your steps and the latest commits is the result.
Hope it's okey now.

castiel652 commented:
Well you forgot to add qualcommax to the second commit

skramstad commented:
I did. Imagine that 👍 New commit with `qualcommax:` prefix.

robimarko commented:
Do 620 and 630 share anything?

skramstad commented:
> Do 620 and 630 share anything?

They share a little bit of code in the dts, like nand, uart5. But since it has +1 lan port and some changes for the LEDs, it was more than I thought at the beginning.

robimarko commented:
Ok, I though they shared more so a common DTS should be used.

robimarko commented:
Can you please fix the formal issues first?

Also, please change the sorting commit title to something like: `qualcommax: ipq807x: alphabetically sort Netgear WAX family`

And then expand on what specifically was done in the commit description.

I will go over the other changes as well.

robimarko commented on package/firmware/ipq-wifi/Makefile:
The git source version needs to be updated to include WAX630 BDF as this version doesn't have it.

skramstad replied:
Hi. By that you mean I need to update this file with the latest `PKG_SOURCE_DATE`, `PKG_SOURCE_VERSION` and `PKG_MIRROR_HASH`?

I found the steps here https://openwrt.org/docs/guide-developer/packages#use_source_repository, but is this the correct way?

```
make package/firmware/ipq-wifi/download V=s
make package/firmware/ipq-wifi/check FIXUP=1 V=s
```
And if this is correct, do you want a new separate commit for this, or just update the existing one with the new data?

robimarko replied:
Yes, make the update like this into a separate commit that happens before you add your device:
https://github.com/openwrt/openwrt/commit/eb8ddf5d5c9df9e3ca97c12ee2e7685ffc4c88be

skramstad replied:
Okey. Sorry about the late response. The NIC on my NUC just went offline, but the new Bios FW fixed it...

New commit added for ipq-wifi.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-wax630.dts:
```diff
@@ -0,0 +1,253 @@
+		compatible = "gpio-leds";
+
+		led_system_red: system-red {
+			default-state = "off";
```
`default-state` should go last

skramstad replied:
Done.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-wax630.dts:
```diff
@@ -0,0 +1,253 @@
+			gpios = <&tlmm 21 GPIO_ACTIVE_LOW>;
+		};
+
+		led_lan1_g {
```
Underscores should not be used as part of node names.

So, please rename this into something like: `lan1-green`

skramstad replied:
Done.

robimarko commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -146,6 +156,22 @@ define Device/netgear_wax620
+	IMAGES := ui-factory.tar factory.ubi sysupgrade.bin
+	IMAGE/ui-factory.tar := append-ubi | wax6xx-netgear-tar
+	DEVICE_PACKAGES += kmod-spi-gpio \
+		ipq-wifi-netgear_wax630
```
No need to break the line for just 2 packages

skramstad replied:
Done.

skramstad committed:
qualcommax: ipq807x: alphabetically sort Netgear WAX family.

As requested by the maintainers, the order for the WAX family
should be alphabetically. WAX620 is now after WAX218.

Files changed:
+ipq807x.mk
+01_leds

No changes to the content.

Signed-off-by: Kristian Skramstad <kristian+github@83.no>

skramstad committed:
ipq-wifi: update to version 2023-10-12

Contains the following update:
52a1c29 ipq8074: add Netgear WAX630 Board file for Netgear WAX630. Extracted from stock (WAX630_BDF.bin) firmware and repacked.
e7701b8 ipq8074: update RegDB in new submitted BDF
cd04ab7 qcn9074: update RegDB in new submitted BDF

Signed-off-by: Kristian Skramstad <kristian+github@83.no>

skramstad committed:
qualcommax: ipq807x: add support for Netgear WAX630

```
Specifications:
* CPU: Qualcomm IPQ8074A, SoC Version: 2.0, Quad core 1651 MHz
* RAM: 1 GiB of DDR3 466 MHz
* Flash: NAND 512 MiB (Winbond W29N04GZ)
* 6 RGB LEDs: Power, LAN1, LAN2, 2.4GHz, 5GHz H and 5GHz L
* UART: One 4-pin populated header next to the heatsink and a chip.
  GND RXD TXD, beginning from the external antennas. 115200n8.

Lan:
* One 100/1000/2.5GBASE-T Gigabit Ethernet 802.3bt/at
* One 100/1000 Gigabit Ethernet

Wlan:
* 4x4 in 2.4GHz  : 802.11b/g/n/ax
* 4x4 in 5.0GHz L: 802.11a/n/ac/ax
* 4x4 in 5.0GHz H: 802.11a/n/ac/ax
* OFDM and OFDMA
* Bidir and MU-MIMO
* Internal antenna 2.86/4.41/4.98 dBi (2.4GHz/5GHz L/5GHz H)

Power:
* 802.3bt/at 30.1W
* DC 12V/3.5A

Mounting: Wall and ceiling
```

```
1.  Download the OpenWrt initramfs image. Copy the image to a TFTP server
2.  Connect to the console on the AP, and connect the LAN port to your LAN
3.  Stop auto boot to get to U-boot shell, interrupt the autoboot process by pressing '0' when prompted
4.  Set active_fw in env
4.  Set active_fw in env
    # setenv active_fw 1
5.  Transfer the initramfs image with TFTP
    # setenv serverip 192.168.1.10 (IP of TFTP server host)
    # setenv ipaddr 192.168.1.1 (IP used by the router for getting the image, must be in the same subnet as the TFTP host)
    # tftpboot openwrt-qualcommax-ipq807x-netgear_wax630-initramfs-uImage.itb
6.  Reboot and load the image
    # bootm
7.  SCP factory image to the AP
    # scp openwrt-qualcommax-ipq807x-netgear_wax630-squashfs-factory.ubi root@192.168.1.1:/tmp/
8.  Connect to device using SSH (use the LAN port)
9.  Flash squashfs-factory.ubi from within the initramfs instance of OpenWRT
    Before you flash, please check your mtd partitions where mtdX is the right mtd rootfs partition.
    # cat /proc/mtd (To check MTD partitions)
    # ubiformat /dev/mtd18 -y -f /tmp/openwrt-qualcommax-ipq807x-netgear_wax630-squashfs-factory.ubi
10. Set active_fw to 0
    # /usr/sbin/fw_setenv active_fw 0
11. Reboot the AP and your done
    # reboot
```

Reviewed-by: Robert Marko <robimarko@gmail.com>
Signed-off-by: Kristian Skramstad <kristian+github@83.no>

Ansuel commented:
@robimarko think this is ready?

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-wax630.dts:
```diff
@@ -0,0 +1,253 @@
+		};
+	};
+
+	led_spi {
```
`led-spi`

skramstad replied:
Ok.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-wax630.dts:
```diff
@@ -0,0 +1,253 @@
+		sck-gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&tlmm 19 GPIO_ACTIVE_HIGH>;
+
+		led_gpio: led_gpio@0 {
```
`led-gpio@0`

skramstad replied:
Ok.

robimarko commented on target/linux/qualcommax/image/ipq807x.mk:
```diff
@@ -146,6 +156,21 @@ define Device/netgear_wax620
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	SOC := ipq8074
+	IMAGES := ui-factory.tar factory.ubi sysupgrade.bin
```
Is the `factory.ubi` ever used?

skramstad replied:
Yes.
`# ubiformat /dev/mtd18 -y -f /tmp/openwrt-qualcommax-ipq807x-netgear_wax630-squashfs-factory.ubi`

robimarko commented:
Can you rebase and force-push this so CI runs again?

If that passes:
`Reviewed-by: Robert Marko <robimarko@gmail.com>`

skramstad commented:
@robimarko, you meant like this? Or did I misunderstand...

robimarko commented:
Yes, that retriggered the CI.

Your `ipq-wifi` package bump is not correct, the hash is incorrect.
Run: `make package/firmware/ipq-wifi/check FIXUP=1` which will fix it for you

skramstad commented:
Done.

robimarko commented:
`Reviewed-by: Robert Marko <robimarko@gmail.com>`

skramstad commented:
Hi @robimarko, quick question before it's to late. 
If I want to use DHCP on LAN1 instead of static, what is the code? 
Since LAN on WAX620 is DHCP, I thought this one could be the same too. 
Ref: `ucidef_set_interface_lan "lan1 lan2"`

Thanks.

robimarko commented:
DHCP on LAN is usually only used if there is a single port, but you can enable it by adding `"dhcp"` like:
`ucidef_set_interface_lan "lan1 lan2" "dhcp"`

You cannot just configure it for one port though by default

skramstad commented:
Great, thanks!

skramstad commented:
Hi @Ansuel. Is the PR good to go?
